lib/xunlink.h

```diff
@@ -52,6 +52,10 @@
  * These wrappers present the same interface as the original functions,
  * but will also log an error if the original function did anything
  * other than succeed or fail-due-to-ENOENT
+ *
+ * If the original function fails with ENOENT, errno will be reset to zero.
+ * This prevents errno leaks in the common case, but it means you can't use
+ * these wrappers if you need to detect ENOENT failures.
  **/
```

Comment:
This should also be adjusted to say that the function will return 0 for ENOENT instead of -1

Reply:
Ahh yeah good catch!

Reply:
I've rewritten this comment quite a bit, what do you think?

Reply:
It would actually be more in keeping with errno's usual behaviour to reset it to its previous value rather than 0 when ENOENT is detected (since a syscall that succeeds does not change errno at all).  I don't think it matters much in practice, but I'm overthinking the details now.